2015/IsConstructor.js

```diff
@@ -1,7 +1,34 @@
 'use strict';
 
+var GetIntrinsic = require('../GetIntrinsic.js');
+
+var $construct = GetIntrinsic('%Reflect.construct%', true);
+var $create = GetIntrinsic('%Object.create%', true);
+
 // https://www.ecma-international.org/ecma-262/6.0/#sec-isconstructor
 
-module.exports = function IsConstructor(argument) {
-	return typeof argument === 'function' && !!argument.prototype; // unfortunately there's no way to truly check this without try/catch `new argument`
-};
+if ($construct && $create) {
```

Comment:
testing for these methods isn't sufficient to ensure getters work; `Object.create` is frequently shammed without support for the second argument.